lib/logstorage/partition.go

```diff
@@ -53,7 +53,7 @@ func mustCreatePartition(path string) {
 //
 // The partition must be closed with MustClose before deleting it.
 func mustDeletePartition(path string) {
-	fs.MustRemoveAll(path)
+	fs.MustRemoveDirAtomic(path)
```

Comment:
The `fs.MustRemoveDirAtomic()` must be avoided because it isn't compatible with object storage semantics and with NFS semantics - e.g. object storage and NFS don't provide an API for atomic move of a "directory" with multiple entries. It is better to leave the call to `fs.MustRemoveAll()`, and add a code, which deletes partially deleted partitions on startup. For example, the parts.json file inside the partition directory could be removed first, and after that the `fs.MustRemoveAll()` could be called on the partition directory. Then the partition directory could be removed on startup if it has no parts.json file.

Cc @AndrewChubatiuk , who works on object storage support.